svelte/src/lib/UplotSvelte.svelte

```diff
@@ -0,0 +1,104 @@
+<script lang="ts">
+    import 'uplot/dist/uPlot.min.css';
+    import type uPlot from 'uplot';
+    import {onDestroy, onMount} from "svelte";
+    // import {optionsUpdateState} from "uplot-wrappers-common";
+
+    let uPlot: uPlot;
+    export let options: uPlot.Options;
+    export let data: uPlot.AlignedData;
+    export let target: HTMLDivElement | null = null;
+    export let onDelete: (chart: uPlot) => void = () => {};
+    export let onCreate: (chart: uPlot) => void = () => {};
+    export let resetScales:boolean = true;
+
+    let chart: uPlot | null = null;
+    let div: HTMLDivElement;
+
+    const destroy = () => {
+        if (chart) {
+            chart.destroy();
+            onDelete(chart);
+            chart = null;
+        }
+    };
+
+    const create = () => {
+        if (uPlot && !chart) {
+            chart = new uPlot(options, data, target || div);
+            onCreate(chart);
+        }
+    };
+
+    onMount(async () => {
+        const uplotModule = await import('uplot');
```

Comment:
Why do we have to import it dynamically? Please, import everything statically, there is no need for it right now.

Reply:
There is unfortunately I problem I ended up with when trying to import it statically. Uplot requests access to the DOM. In svelte it could be that the code you are running can be still in the server and the document still not loaded. By using `onMount` you would ensure that the code inside is ran when the document object is accessible. 

To make it accessible to Svelte users (and not only SvelteKit) it is not possible to use SvelteKit's elegant way of checking this - [docs entry for best practices](https://kit.svelte.dev/docs/packaging#best-practices).

The other alternative would be to use the package [esm-env](https://github.com/benmccann/esm-env). However, I am not sure if you'd agree with adding more dependencies.

Reply:
Hmm, that is strange. uPlot shouldn't access DOM before creating an instance of it. So I though it should work just fine... Could you share the error that you have encountered there?

Reply:
```
document is not defined

ReferenceError: document is not defined
    at Object.<anonymous> (/home/.../Projects/uplot-wrappers/node_modules/uplot/dist/uPlot.cjs.js:487:13)
    at Module._compile (node:internal/modules/cjs/loader:1378:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1437:10)
    at Module.load (node:internal/modules/cjs/loader:1212:32)
    at Module._load (node:internal/modules/cjs/loader:1028:12)
    at cjsLoader (node:internal/modules/esm/translators:359:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:308:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async nodeImport (file:///home/.../Projects/uplot-wrappers/node_modules/vite/dist/node/chunks/dep-cNe07EU9.js:55096:17
    ```
    
    Essentially `document` not found. I believe this is a common issue for Svelte + uPlot and one has to use either workaround for it.